ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/validator/ExpressionValidator.java

```diff
@@ -42,20 +43,11 @@ public void setNextValidator(Validator nextValidator) {
     @Override
     public boolean evaluate(String source) {
         IncompleteInputFinder incompleteInputFinder = new IncompleteInputFinder();
-        boolean isIncomplete = NodeParser.parseExpression(source).apply(incompleteInputFinder);
-        if (!NodeParser.parseExpression(source).hasDiagnostics()) {
-            return false;
-        } else {
-            if (!source.endsWith(";") && !NodeParser.parseExpression(source + ";").hasDiagnostics()) {
-                return false;
-            }
-        }
-
-        if (!isIncomplete) {
-            return nextInValidator.evaluate(source);
-        } else {
+        Node parsedNode = NodeParser.parseExpression(source);
+        if (!parsedNode.hasDiagnostics() || !parsedNode.apply(incompleteInputFinder)) {
             return true;
         }
 
+        return nextInValidator.evaluate(source);
```

Comment:
Can't we use `return !parsedNode.hasDiagnostics() || !parsedNode.apply(incompleteInputFinder) || nextInValidator.evaluate(source)`

Reply:
Updated in ea2b94a0c6e70886dde225592eb0985699ae6232